src/macros.rs

```diff
@@ -715,6 +715,82 @@ macro_rules! query_file_scalar_unchecked (
+/// or when you do `cargo run` your application isn't applying new migrations on startup.
+///
+/// This is because our ability to tell the compiler to watch external files for changes
+/// from a proc-macro is very limited. The compiler by default only re-runs proc macros when
+/// one ore more source files have changed, because normally it shouldn't have to otherwise. SQLx is
+/// just weird in that external factors can change the output of proc macros, much to the chagrin of
+/// the compiler team and IDE plugin authors.
+///
+/// As of 0.5.6, we emit `include_str!()` with an absolute path for each migration, but that
+/// only works to get the compiler to watch _existing_ migration files for changes.
+///
+/// Our only options for telling it to watch the whole `migrations/` directory are either via the
+/// user creating a Cargo build script in their project, or using an unstable API on nightly
+/// governed by a `cfg`-flag.
+///
+/// ##### Stable Rust: Cargo Build Script
+/// The only solution on stable Rust right now is to create a Cargo build script in your project
+/// and have it print `cargo:rerun-if-changed=migrations`:
+///
+/// `build.rs`
+/// ```
+/// fn main() {
+///     println!("cargo:rerun-if-changed=migrations");
+/// }
+/// ```
+///
+/// You can also do this automatically by running `sqlx migrate build-script`.
+///
+/// See: [The Cargo Book: 3.8 Build Scripts; Outputs of the Build Script](https://doc.rust-lang.org/stable/cargo/reference/build-scripts.html#outputs-of-the-build-script)
+///
+/// #### Nightly Rust: `cfg` Flag
+/// The `migrate!()` macro also listens to `--cfg procmacro2_semver_exempt`, which will enable
+/// the `track_path` feature to directly tell the compiler to watch the `migrations/` directory:
+///
+/// ```sh,ignore
+/// $ env RUSTFLAGS='--cfg procmacro2_semver_exempt' cargo build
+/// ```
+///
+/// Note that this unfortunately will trigger a fully recompile of your dependency tree, at least
+/// for the first time you use it.
+///
+/// You can also set it in `build.rustflags` in `.cargo/config.toml`:
+/// ```toml,ignore
+/// [build]
+/// rustflags = ["--cfg procmacro2_semver_exempt"]
+/// ```
+///
+/// And then continue building and running your project normally.
+///
+/// If you're building on nightly anyways, it would be extremely helpful to help us test
+/// this feature and find any bugs in it.
+///
+/// Subscribe to [the `track_path` tracking issue](https://github.com/rust-lang/rust/issues/73921)
+/// for discussion and the future stabilization of this feature.
+///
+/// **Why not a feature flag?** Feature flags enabling nightly features are technically an anti-pattern
+/// because they take away agency from the person compiling the final binary, even though
+/// the choice of stable vs nightly _should_ be up to them. If you're using SQLx in a library, you
+/// should not set this flag but instead direct your end-users to use it, if applicable.
```

Comment:
This explanation doesn't make much sense to me. I can see that it would generally be a bad idea to have library authors enable this (though I'm wondering whether it will even take effect for reverse deps with the `.cargo/config` based solution), but if things only work reliably with that feature, is it really any better to require downstream to enable the cfg, rather than documenting that the library crate only works with nightly?

Reply:
It's also not really necessary with a library because this whole issue only arises if you're adding migration files without changing Rust source files, which you would only be doing if you're building from a local source instead of crates.io.

So it also doesn't really make sense to be a feature flag because it's _only_ useful to the end-user, or else someone who's making modifications to the library and running tests or something.

I'm decently sure `.cargo/config.toml` is explicitly _not_ published with the crate because it's meant for local tweaks like this, and yes it should affect reverse deps because it's the same as setting the `RUSTFLAGS` env var. In fact, it's documented like this:

```toml
[build]
...
rustflags = ["…", "…"]        # custom flags to pass to *all* compiler invocations
```
(Emphasis mine.)

Reply:
I got rid of the justification in the docs cause it made them too long.